ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -491,6 +491,7 @@ private void runReAct(
                                     llmToolTmpParameters.put(MLAgentExecutor.QUESTION, actionInput);
                                     tools.get(action).run(llmToolTmpParameters, toolListener); // run tool
                                 } else {
+                                    toolParams.putAll(toolSpecMap.get(action).getParameters());
```

Comment:
do we need to take `input` as special case here, if toolSpecMap contains `input`, that will override the value from LLM?

Reply:
Makes sense, moved it ahead of input part.